src/eradiate/scenes/surface/_dem.py

```diff
@@ -83,224 +260,250 @@ class DEMSurface(Surface):
+        Evolve the DEM to include the atmospheric geometry.
 
-        bsdf : .BSDF
-            BSDF to be attached to the mesh shape.
+        In case of a spherical shell geometry, the DEM will follow the curvature of the sphere.
+        """
+        theta_lim, phi_lim = minmax_coordinates(self.shape.vertices)
+        theta_mean, phi_mean = mean_coordinates(self.shape.vertices)
+        try:
+            atmo_bottom = geometry.atmosphere_bottom
+        except AttributeError:
+            atmo_bottom = 0 * ureg.m
+
+        if os.path.isfile(self.mesh_filename):
+            warnings.warn(
+                f"Mesh file with filename {self.mesh_filename} exists. Skipping creation of DEM mesh file."
+            )
+        else:
+            if isinstance(geometry, SphericalShellGeometry):
+                trafo = (
+                    mi.Transform4f.rotate(axis=[0, 0, 1], angle=-90)
+                    @ mi.Transform4f.rotate(axis=[0, 1, 0], angle=-(90 - theta_mean))
+                    @ mi.Transform4f.rotate(axis=[0, 0, 1], angle=-phi_mean)
+                )
+                vertices = self.shape.vertices
+                vertices_spherical = transform_vertices_spherical_shell(
+                    vertices, geometry.planet_radius
+                )
+                vertices_new = [trafo @ vertex for vertex in vertices_spherical]
+            elif isinstance(geometry, PlaneParallelGeometry):
+                vertices = self.shape.vertices
+                vertices_new = transform_vertices_plane_parallel(
+                    vertices,
+                    planet_radius=6378 * ureg.km,
+                    altitude=atmo_bottom,
+                )
+            self.shape = attrs.evolve(self.shape, vertices=vertices_new)
+            mesh = self.shape.instance
+            mesh.write_ply(str(self.mesh_filename))
+
+        if isinstance(geometry, SphericalShellGeometry):
+            opacity_mask_trafo = _to_uv(*theta_lim, *phi_lim)
+            opacity_array = np.ones((3, 3))
+            opacity_array[1, 1] = 0
+            opacity_bitmap = mi.Bitmap(opacity_array)
+
+            opacity_bsdf = OpacityMaskBSDF(
+                id=f"{self._bsdf_id}_background",
+                inner_bsdf=self.bsdf,
+                opacity_bitmap=opacity_bitmap,
+                uv_trafo=opacity_mask_trafo,
+            )
+
+            trafo = (
+                mi.Transform4f.rotate(axis=[0, 0, 1], angle=-90)
+                @ mi.Transform4f.rotate(axis=[0, 1, 0], angle=(90 - theta_mean))
+                @ mi.Transform4f.rotate(axis=[0, 0, 1], angle=-phi_mean)
+            )
+            surface_background = SphereShape.surface(
+                id=f"{self.id}_background",
+                altitude=atmo_bottom,
+                planet_radius=geometry.planet_radius,
+                to_world=trafo,
+                bsdf=opacity_bsdf,
+            )
+        elif isinstance(geometry, PlaneParallelGeometry):
+            planet_radius = 6378 * ureg.km
```

Comment:
Please use `geometry.planet_radius`

Reply:
I can't. This is the PlaneParallelGeometry, which does not hold a planet radius. Maybe I should explain my approach a little: When the dem class is instantiated, a BufferMeshShape is created, which holds the vertex positions given as (latitude, longitude, elevation) tuples, in [degrees, degrees, ucc["length"]] units. from that, I convert to cartesian coordinates later, in the `evolve_dem` method. Therefore, I need a planet radius even in the plane parallel case, in order to convert lat/lon coordinates to cartesian. This approach is as hacky as any other idea I had for the dem...

Reply:
Right, I misread; then I think this should be tunable and default to `eradiate.constants.EARTH_RADIUS`.